### PR TITLE
Fix issues reported by Coverity

### PR DESCRIPTION
**Pull request by Danielius1922**
## Summary by CodeRabbit

- **New Features**
	- Enhanced introspection data handling, allowing for data to be read from a file and set for devices more efficiently.

- **Refactor**
	- Improved error reporting in introspection data functions.
	- Streamlined input handling in cloud certification tests.

- **Bug Fixes**
	- Addressed potential buffer overflow in trust anchor provisioning by switching to safer string copying methods.

### CODE REVIEW

File: onboarding_tool/obtmain.c [hunk 1]
```diff
@@ -1647,12 +1646,19 @@ provision_ace2(void)
     return;
   }
 
-  while (num_resources <= 0 || num_resources > MAX_NUM_RESOURCES) {
-    if (num_resources != 0) {
+  int num_resources = 0;
+  while (true) {
+    OC_PRINTF("\nEnter number of resources in this ACE: ");
+    if (scanf("%d", &num_resources) <= 0) {
```
Comment (jkralik) [inline onboarding_tool/obtmain.c:1652]: Is it correct to not use `SCANF` macro ?

File: apps/simpleserver-TVAppAndAction.c [hunk 2]
```diff
@@ -54,12 +54,54 @@ static oc_string_array_t my_supportedactions;
...
+  }
+  ret = ftell(fp);
+  if (ret < 0) {
+    fclose(fp);
+    return false;
+  }
+  rewind(fp);
+
+  size_t buffer_size = (size_t)ret;
+  uint8_t *buffer = (uint8_t *)malloc(buffer_size * sizeof(uint8_t));
```
Comment (jkralik) [inline apps/simpleserver-TVAppAndAction.c:78]: malloc can return `NULL`

File: apps/cloud_proxy.c [hunk 3]
```diff
@@ -425,6 +425,44 @@ is_udn_listed(char *udn)
...
+  }
+  ret = ftell(fp);
+  if (ret < 0) {
+    fclose(fp);
+    return false;
+  }
+  rewind(fp);
+
+  size_t buffer_size = (size_t)ret;
+  uint8_t *buffer = (uint8_t *)malloc(buffer_size * sizeof(uint8_t));
```
Comment (jkralik) [inline apps/cloud_proxy.c:449]: malloc can return `NULL`

File: apps/client_certification_tests.c [hunk 4]
```diff
@@ -84,43 +84,61 @@ free_all_resources(void)
...
+  }
+  ret = ftell(fp);
+  if (ret < 0) {
+    fclose(fp);
+    return false;
+  }
+  rewind(fp);
+
+  size_t buffer_size = (size_t)ret;
+  uint8_t *buffer = (uint8_t *)malloc(buffer_size * sizeof(uint8_t));
```
Comment (jkralik) [inline apps/client_certification_tests.c:108]: malloc can return `NULL`

File: apps/smart_home_server_linux.c [hunk 5]
```diff
@@ -82,6 +85,42 @@ toggle_switch_resource(void *data)
...
+  }
+  ret = ftell(fp);
+  if (ret < 0) {
+    fclose(fp);
+    return false;
+  }
+  rewind(fp);
+
+  size_t buffer_size = (size_t)ret;
+  uint8_t *buffer = (uint8_t *)malloc(buffer_size * sizeof(uint8_t));
```
Comment (jkralik) [inline apps/smart_home_server_linux.c:109]: malloc can return `NULL`

**Comment by jkralik** [comment]
@coderabbitai review

File: onboarding_tool/obtmain.c (hunk 1 above)
Comment (Danielius1922) [reply onboarding_tool/obtmain.c:1652]: Yes, it's fine, `SCANF` is just a local macro defined in this file that logs error and continues execution:

```C
#define SCANF(...)                                                             \
  do {                                                                         \
    if (scanf(__VA_ARGS__) <= 0) {                                             \
      OC_PRINTF("ERROR Invalid input\n");                                      \
      fflush(stdin);                                                           \
    }                                                                          \
  } while (0)
```

[the remark this answers is not in the document]
Comment (Danielius1922) [reply onboarding_tool/obtmain.c:1652]: @coderabbitai Check your previous comments and resolve fixed.

**Pull request merged**